src/searching/linear_search.rs

```diff
@@ -0,0 +1,131 @@
+/// ## Linear Search algorithm.
+/// Iterating over the array linearly and returns the index value of the target element in the array.
```

Comment:
"Performs a linear search over a collection by iterating over its elements and returning the element's index if found or `None`."